### DISCUSSION TITLE
How to use @DgsData.List for returning multiple fields

### DISCUSSION THREAD

**Discussion by BCantos17** [Netflix/dgs-framework#899] [Q&A] [answered]
Currently have a graphql object returning fields from different sources and I'm looking to leverage `@DgsData.List`. I found this https://github.com/Netflix/dgs-framework/pull/574 and this https://netflix.github.io/dgs/datafetching/ (under Multiple @DgsData Annotations Via @DgsData.List) but these don't tell me how to return back different results using DgsData.List

For example, if we have this

```graphql
type Query {
  movies: [Movie]
  shows: [Show]
}
```

I would like to resolve movies and shows somehow like this
```kotlin
@DgsData.List(
    DgsData(parentType = "Query", field = "movies"),
    DgsData(parentType = "Query", field = "shows")
)
fun getMoviesAndShow(dfe: DataFetchingEnviornment): Query {
   val query = repo.getMoviesAndShows()
   return query // contains movies and shows
}
``` 

Is something like this possible?

**Comment by srinivasankavitha** [comment, marked as answer]
With @DgsData.List, you can use the same data fetcher for resolving
`movies` and `shows` independently. In other words, when `graphql-java` is
resolving `movies` or `shows` it will invoke this data fetcher. It will not
resolve both at the same time. Hope that answers your question.

**Reply by BCantos17** [reply]
I see, so it will call the data fetcher twice in my example. I suppose I could implement caching in order to avoid calling the same data source twice.

Still I think it might be a good feature to have, having two data fetcher for resolving multiple different fields in one query, considering the nature of graphql. Has this ever been discussed?

**Comment by srinivasankavitha** [comment]
As you mentioned, this should be a caching concern handled at the
application layer. If you know that both movies and shows are going to be
fetched at some point, then the data fetcher should be optimized to
return the cached value if the data already exists. Fundamentally,
`graphql-java` needs to know what data fetcher to call for a field. It is
up to the user to implement the same shared data fetcher or distinct ones.
